### PR TITLE
Thief pinpointer

### PR DESCRIPTION
**Pull request by TheShuEd**
## About the PR

added a thief's pinpointer! It allows you to track the location of targets that are on the antagonist's task steal list.
A thief can get it in the new kit, and it can also be bought in uplink for 3 TCs

New robbers kit:
- Thief pinpointer
- x2 storage implanter
- Thief toy figurine

## Media

https://github.com/space-wizards/space-station-14/assets/96445749/812abcea-7e98-4b23-a002-8fc4a91d142a

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
:cl:
- add: added thief's pinpointer in uplink. 3 TC
- add: added new Thief kit: Thief pinpointer and x2 Storage implanter

### CODE REVIEW

**Comment by lzk228** [comment]
OP?

**Comment by TheShuEd** [comment]
suggest balance changes

**Comment by Ilya246** [comment]
i think it's fine, it prevents heads trolling and hiding their objectives and also lets you know where to find a certain item but does not provide any other benefit

**Comment by TheShuEd** [comment]
It might make sense to increase the cost.in uplink, or change the contents of the set.  I'm still thinking.

**Comment by Dutch-VanDerLinde** [comment]
It's a good uplink price

**Comment by TheShuEd** [comment]
requied #23366

**Comment by TheShuEd** [comment]
is no longer https://github.com/space-wizards/space-station-14/pull/23366 required. You can do a review.

File: Content.Server/Pinpointer/PinpointerSystem.cs [hunk 1]
```diff
@@ -22,6 +34,54 @@ public override void Initialize()
 
         SubscribeLocalEvent<PinpointerComponent, ActivateInWorldEvent>(OnActivate);
         SubscribeLocalEvent<FTLCompletedEvent>(OnLocateTarget);
+        SubscribeLocalEvent<AntagPinpointerComponent, GetVerbsEvent<Verb>>(OnGetVerb);
+    }
+
+    private void OnGetVerb(Entity<AntagPinpointerComponent> antagPinpointer, ref GetVerbsEvent<Verb> args)
+    {
+        if (!args.CanAccess || !args.CanInteract || args.Hands == null)
```
Comment (Jezithyr) [inline Content.Server/Pinpointer/PinpointerSystem.cs:42]: Combine these, there is no reason to have this many separate if statements that all return the same thing

File: Content.Server/Pinpointer/PinpointerSystem.cs [hunk 2]
```diff
@@ -47,6 +107,9 @@ private void OnActivate(EntityUid uid, PinpointerComponent component, ActivateIn
     {
         TogglePinpointer(uid, component);
 
+        if (HasComp<AntagPinpointerComponent>(uid))
```
Comment (Jezithyr) [inline Content.Server/Pinpointer/PinpointerSystem.cs:110]: Why is this needed?

File: Content.Server/Pinpointer/PinpointerSystem.cs [hunk 3]
```diff
@@ -127,14 +200,47 @@ public override void Update(float frameTime)
         return l.Count > 0 ? l.First().Value : null;
     }
 
+    /// <summary>
+    ///     Try to find the closest entity from stealGroup on a current map
+    ///     Will return null if can't find anything
+    /// </summary>
+    private EntityUid? FindTargetFromStealGroup(EntityUid uid, string group)
+    {
+        foreach (var target in _lookup.GetEntitiesInRange<StealTargetComponent>(_transform.GetMapCoordinates(uid), 100))
```
Comment (Jezithyr) [inline Content.Server/Pinpointer/PinpointerSystem.cs:209]: Don't make unconstrained entity searches. This gets every single entity within your range (which will be alot). Use a component bounded search instead since this only searches entities that have the specified component

File: Content.Shared/Pinpointer/AntagPinpointerComponent.cs
Comment (Jezithyr) [inline Content.Shared/Pinpointer/AntagPinpointerComponent.cs:1]: This doesn't need to exist

**Review by Jezithyr — CHANGES_REQUESTED** [review]

**Comment by Arendian** [comment]
5 of your files have trailing whitespaces.

I like the concept, getting this pinpointer will make thief gameplay more about actively trying to steal the stuff you need, compared to currently spending most of your time finding your target.
After my changes in #23555 I think only our OnVerb methods and the way we handle popups will be conflicting, these conflicts will be trivial to solve.

**Comment by TheShuEd** [comment]
I finish it after #23555
